crates/stc_ts_file_analyzer/src/analyzer/expr/call_new.rs

```diff
@@ -2953,14 +3028,25 @@ impl Analyzer<'_, '_> {
         Ok(ret_ty)
     }
 
-    fn validate_arg_types(&mut self, params: &[FnParam], spread_arg_types: &[TypeOrSpread], is_generic: bool) {
+    fn validate_arg_types(
+        &mut self,
+        params: &[FnParam],
+        spread_arg_types: &[TypeOrSpread],
+        is_generic: bool,
+        is_overload: bool,
+    ) -> VResult<()> {
         info!("[exprs] Validating arguments");
 
         macro_rules! report_err {
             ($err:expr) => {{
-                self.storage.report($err.context("tried to validate an argument"));
+                if is_overload {
+                    return Err($err.into());
+                } else {
+                    self.storage.report($err.context("tried to validate an argument"));
+                }
```

Comment:
This isn't great but I couldn't find a better way while not changing the `validate_arg_types` too much

Reply:
Which way do you prefer?
I don't have a strong opinion here

Reply:
You mean whether to block the PR or not? Honestly I think blocking the PR is better because there might be something I havent accounted for and this PR will likely  require changes. It also gives motivation to solve the other issue.

Reply:
Thank you! Then I'll wait 👍